py/objstr.c

```diff
@@ -41,6 +41,23 @@ STATIC mp_obj_t str_modulo_format(mp_obj_t pattern, size_t n_args, const mp_obj_
 STATIC mp_obj_t mp_obj_new_bytes_iterator(mp_obj_t str, mp_obj_iter_buf_t *iter_buf);
 STATIC NORETURN void bad_implicit_conversion(mp_obj_t self_in);
 
+STATIC void mp_check_compat_arg(const mp_obj_type_t *self_type, const mp_obj_t arg) {
```

Comment:
As it's a static helper, shouldn't be called `mp_`. Maybe just `str_check_arg_type` or something.

Reply:
Maybe a quick comment just to explain what's going on here:

`// Arguments to string operations need to be the same type (e.g. str.find(str), with the exception that bytes may be used for bytearray and vice versa."